plugins/SlewDistortion/SlewDistortionControlDialog.cpp

```diff
@@ -0,0 +1,479 @@
+	bias2Draggable->setDefaultValPixmap(PLUGIN_NAME::getIconPixmap("handle_zero"));
+	
+	m_slewUp1Knob = makeKnob(96, 65, tr("Slew Up 1:"), "", &controls->m_slewUp1Model);
+	m_slewUp2Knob = makeKnob(96, 186, tr("Slew Up 2:"), "", &controls->m_slewUp2Model);
+	m_slewDown1Knob = makeKnob(163, 65, tr("Slew Down 1:"), "",
+		controls->m_slewLink1Model.value() ? &controls->m_slewUp1Model : &controls->m_slewDown1Model);
+	m_slewDown2Knob = makeKnob(163, 186, tr("Slew Down 2:"), "",
+		controls->m_slewLink2Model.value() ? &controls->m_slewUp2Model : &controls->m_slewDown2Model);
+	makeKnob(329, 26, tr("Warp 1:"), "", &controls->m_warp1Model);
+	makeKnob(329, 147, tr("Warp 2:"), "", &controls->m_warp2Model);
+	makeKnob(371, 26, tr("Crush 1:"), "", &controls->m_crush1Model);
+	makeKnob(371, 147, tr("Crush 2:"), "", &controls->m_crush2Model);
+	makeKnob(225, 65, tr("Attack 1:"), "", &controls->m_attack1Model);
+	makeKnob(225, 186, tr("Attack 2:"), "", &controls->m_attack2Model);
+	makeKnob(267, 65, tr("Release 1:"), "", &controls->m_release1Model);
+	makeKnob(267, 186, tr("Release 2:"), "", &controls->m_release2Model);
+	makeKnob(225, 26, tr("Dynamics 1:"), "", &controls->m_dynamics1Model);
+	makeKnob(225, 147, tr("Dynamics 2:"), "", &controls->m_dynamics2Model);
+	makeKnob(267, 26, tr("Dynamic Slew 1:"), "", &controls->m_dynamicSlew1Model);
+	makeKnob(267, 147, tr("Dynamic Slew 2:"), "", &controls->m_dynamicSlew2Model);
+	
+	Draggable* outVol1Draggable = new Draggable(FloatModelEditorBase::DirectionOfManipulation::Vertical,
+		&controls->m_outVol1Model, PLUGIN_NAME::getIconPixmap("handle"), 108, 34, this);
+	outVol1Draggable->move(594, outVol1Draggable->y());
+	outVol1Draggable->setDefaultValPixmap(PLUGIN_NAME::getIconPixmap("handle_zero"));
+	Draggable* outVol2Draggable = new Draggable(FloatModelEditorBase::DirectionOfManipulation::Vertical,
+		&controls->m_outVol2Model, PLUGIN_NAME::getIconPixmap("handle"), 229, 155, this);
+	outVol2Draggable->move(594, outVol2Draggable->y());
+	outVol2Draggable->setDefaultValPixmap(PLUGIN_NAME::getIconPixmap("handle_zero"));
+	
+	PixmapButton* slewLink1Button = makeToggleButton(132, 70, tr("Slew Link 1"), "link_on", "link_off", &controls->m_slewLink1Model);
+	connect(slewLink1Button, &PixmapButton::clicked, this, [this, controls]{
+		controls->m_slewDown1Model.setValue(controls->m_slewUp1Model.value());
+		m_slewDown1Knob->setModel(controls->m_slewLink1Model.value() ? &controls->m_slewUp1Model : &controls->m_slewDown1Model);
+	}, Qt::DirectConnection);
+	PixmapButton* slewLink2Button = makeToggleButton(132, 191, tr("Slew Link 2"), "link_on", "link_off", &controls->m_slewLink2Model);
+	connect(slewLink2Button, &PixmapButton::clicked, this, [this, controls]{
+		controls->m_slewDown2Model.setValue(controls->m_slewUp2Model.value());
+		m_slewDown1Knob->setModel(controls->m_slewLink1Model.value() ? &controls->m_slewUp1Model : &controls->m_slewDown1Model);
+	}, Qt::DirectConnection);
```

Comment:
I ran into the issue that he lower band slew rate knobs cannot be unlinked. I think these need to be changed?
```suggestion
	PixmapButton* slewLink2Button = makeToggleButton(132, 191, tr("Slew Link 2"), "link_on", "link_off", &controls->m_slewLink2Model);
	connect(slewLink2Button, &PixmapButton::clicked, this, [this, controls]{
		controls->m_slewDown2Model.setValue(controls->m_slewUp2Model.value());
		m_slewDown2Knob->setModel(controls->m_slewLink2Model.value() ? &controls->m_slewUp2Model : &controls->m_slewDown2Model);
	}, Qt::DirectConnection);
```

Reply:
Oops, somehow I never tested the bottom ones...